server/src/main/java/org/elasticsearch/common/lucene/uid/PerThreadIDVersionAndSeqNoLookup.java

```diff
@@ -96,6 +96,7 @@ final class PerThreadIDVersionAndSeqNoLookup {
         this.loadedTimestampRange = loadTimestampRange;
         if (loadTimestampRange) {
             PointValues tsPointValues = reader.getPointValues(DataStream.TimestampField.FIXED_TIMESTAMP_FIELD);
+            assert tsPointValues != null : "no timestamp field for reader:" + reader + " and parent:" + reader.getContext().parent.reader();
```

Comment:
I have a hunch that we're maybe dealing with a segment that doesn't have actual data in it. But maybe a delete or noop operation, these trans log ops get stored as well in the index and are used for shard recovery and CCR. If that is the case then we simply need add line 100-101 into a not null if check.